virttest/virsh.py

```diff
@@ -1577,8 +1577,16 @@ def undefine(name, options=None, **dargs):
     :return: CmdResult object
     """
     cmd = "undefine %s" % name
-    if options is not None:
-        cmd += " %s" % options
+
+    vmxml = dumpxml(name).stdout_text
+    if "nvram" in vmxml:
+        if options is None or "--nvram" not in options:
+            cmd += " --nvram"
+        else:
+            cmd += " %s" % options
+    else:
+        if options is not None:
+            cmd += " %s" % options
 
     if platform.machine() == "aarch64":
```

Comment:
I wonder if we should remove platform checking here to make it work for all archs?

Reply:
Or add x86_64? I'm not quite sure about other archs, @dzhengfy  what's your opinion?